Feature/client response (#39)
AitoDotAI/aito-python-tools: docs/source/changelog.rst

Changelog
=========

0.4.0
-----

This version contains changes on how you make requests with the :py:class:`~aito.client.AitoClient`.

> In addition to specifying the request method, endpoint as before, you can now use different

Reviewer: ```suggestion
In addition to specifying the request method and endpoint as before, you can now use different
```

:py:mod:`Request objects <aito.client_request>` or use the endpoint methods of the client,
i.e: :py:func:`aito.client.AitoClient.predict`. The client now returns the enriched
:py:mod:`Response objects <aito.client_response>` instead of JSON as before.

Helper methods of the AitoClient, e.g: create_table, are moved to the :py:mod:`aito.api` module.

> You can now execute requests from the CLI

Reviewer: ```suggestion
You can now execute requests from the CLI.
```

SDK
^^^

> - Schema objects validation in the :py:mod:`aito.schema` module is improved

Reviewer: ```suggestion
- Schema objects validation in the :py:mod:`aito.schema` module is improved.
```

- Added the :py:mod:`aito.client_request` module which contains the request objects for :py:class:`~aito.client.AitoClient`.
- Added the :py:mod:`aito.client_response` module which contains the response objects returned by the :py:class:`~aito.client.AitoClient`.

AitoClient
""""""""""

>   - The **AitoClient.request** method no longer takes positional argument. You now have to specify either `method`, `endpoint` or `request_obj`.

Reviewer: ```suggestion
  - The **AitoClient.request** method no longer takes positional arguments. You now have to specify either `request_obj` or `method` and `endpoint`.
```

>   - Both **AitoClient** and **AitoClient.request** now has a `raise_for_status` argument which controls whether the client should raise or return an **aito_client.RequestError** object when an error occurred during sending a request.

Reviewer: ```suggestion
  - Both **AitoClient** and **AitoClient.request** now have a `raise_for_status` argument which controls whether the client should raise or return an **aito_client.RequestError** object when an error occurs during sending a request.
```
